submissions/misshatsker/oop-exercise/index.js

```diff
@@ -0,0 +1,62 @@
+class MyInhabitant {
+   static inhabitantProperties = ['species', 'name', 'gender', 'saying'];
+   static delimiter = '; ';
+
+   constructor(species, name, gender, saying) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.saying = saying;
+      this.properties = [...MyInhabitant.inhabitantProperties];
+   }
+
+   getInhabitantInfo() {
+      const values = this.properties.map((key) => this[key]);
+      return values.join(MyInhabitant.delimiter);
+   }
+}
+
+class Human extends MyInhabitant {
+   constructor(name, gender, saying) {
+      super('human', name, gender, saying);
+      this.hands = 2;
+      this.legs = 2;
+      this.properties.push('hands', 'legs');
+   }
+}
+
+class Animal extends MyInhabitant {
+   constructor(name, gender, saying) {
+      super('animal', name, gender, saying);
+      this.legs = 4;
```

Comment:
Why `legs` property is treated in way different than e.g. `name`?

Reply:
Because `MyInhabitant` doesn't have `legs` property, so I adjusted supplemental behavior in a child class, shouldn't I?

Reply:
Why `MyInhabitant` doesn't have `legs` property?
We can take out `name` from the base class to child classes and say "...because base class doesn't have this property".

Reply:
Oh, I got you, both `Animal` and `Human` contain the `legs` property, so we can move in into a base class as well. I just got confused about property names. I mean how we name it for animals (legs/limbs/paws). My bad, thanks!